onnxruntime/core/providers/dml/DmlExecutionProvider/src/Operators/DmlOperatorQLinearConcat.cpp

```diff
@@ -0,0 +1,235 @@
+
+        std::vector<DML_TENSOR_DESC> inputDescs = GetDmlInputDescs();
+        std::vector<DML_TENSOR_DESC> outputDescs = GetDmlOutputDescs();
+
+        // 1. output edges between Dequantize and Join node
+        // 2. input edge between Join and Quantize node
+        std::vector<TensorDesc> intermediateOutputTensorDescs(input_count);
+        std::vector<DML_TENSOR_DESC> namedDequantizeOperatorDescs(input_count);
+        std::vector<DML_ELEMENT_WISE_DEQUANTIZE_LINEAR_OPERATOR_DESC> dequantizeOperatorDescs(input_count);
+        std::vector<DML_OPERATOR_DESC> dmlOpDesc(input_count);
+        std::vector<const DML_OPERATOR_DESC*> opDescs;
+        for (uint32_t input_index = 0; input_index < input_count; ++input_index)
+        {
+            auto tuple_start = 2 + input_index * 3;
+            intermediateOutputTensorDescs[input_index] = TensorDesc(
+                MLOperatorTensorDataType::Float,
+                kernelCreationContext.GetTensorShapeDescription().GetInputTensorShape(tuple_start),
+                kernelCreationContext.GetTensorShapeDescription().GetInputTensorShape(tuple_start),
+                TensorAxis::DoNotCoerce,
+                TensorAxis::W,
+                TensorAxis::RightAligned,
+                NchwDimensionCount, // minDimensionCount
+                0 // guaranteedBaseOffsetAlignment)
+            );
+            namedDequantizeOperatorDescs[input_index] = intermediateOutputTensorDescs[input_index].GetDmlDesc();
+
+            dequantizeOperatorDescs[input_index].InputTensor = &inputDescs[tuple_start];
+            dequantizeOperatorDescs[input_index].ScaleTensor = &inputDescs[tuple_start + 1];
+            dequantizeOperatorDescs[input_index].ZeroPointTensor = &inputDescs[tuple_start + 2];
+            dequantizeOperatorDescs[input_index].OutputTensor = &namedDequantizeOperatorDescs[input_index];
+
+            dmlOpDesc[input_index] = {DML_OPERATOR_ELEMENT_WISE_DEQUANTIZE_LINEAR, &dequantizeOperatorDescs[input_index]};
+            opDescs.push_back(&dmlOpDesc[input_index]);
+        }
+
+        TensorDesc joinOutputTensorDesc = TensorDesc(
+                MLOperatorTensorDataType::Float,
+                outputShape,
+                outputShape,
+                TensorAxis::DoNotCoerce,
+                TensorAxis::W,
+                TensorAxis::RightAligned,
+                NchwDimensionCount, // minDimensionCount
+                0 // guaranteedBaseOffsetAlignment)
+            );
+        DML_TENSOR_DESC namedJoinOutputTensorDesc = joinOutputTensorDesc.GetDmlDesc();
+
+        DML_JOIN_OPERATOR_DESC joinDesc = {};
+        joinDesc.InputCount = gsl::narrow_cast<uint32_t>(namedDequantizeOperatorDescs.size());
+        joinDesc.InputTensors = namedDequantizeOperatorDescs.data();
+        joinDesc.OutputTensor = &namedJoinOutputTensorDesc;
+        joinDesc.Axis = dmlAxis;
+
+        const DML_OPERATOR_DESC opJoinDesc{DML_OPERATOR_JOIN, &joinDesc};
```

Comment:
```suggestion
        const DML_OPERATOR_DESC opJoinDesc = {DML_OPERATOR_JOIN, &joinDesc};
```

[nit] Consistent assignment style with `        MLOperatorGraphDesc operatorGraphDesc = {};`.